qiskit/transpiler/target.py

```diff
@@ -211,6 +211,21 @@ class Target(BaseTarget):
         an existing object and create a new subset (or use one of the methods
         to do this). The object internally caches different views and these
         would potentially be invalidated by removals.
+
+    Subclassing
+    -----------
+
+    While it is technically possible to subclass :class:`Target`, beware that the majority of the
+    built-in information is in Rust and is queried from Rust in built-in transpiler passes.
+    Python-space overrides are not visible to Rust, and you should not rely on these to change the
+    behavior of Qiskit's built-in transpiler passes.  In general, you should prefer to provide an
+    API to your users based around constructor functions to the standard :class:`Target`.
```

Comment:
I think the only suggestion for ever using a subclass was to provide custom attributes not to override the builtin methods or data (following the classification of required, optional, and custom). I think I understand what you mean here with constructor functions but maybe call it out a bit more explicitly to make the suggestion more clear.